src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -63,8 +63,9 @@ public bool IsSingleSegment
         private ReadOnlySequence(object startSegment, int startIndexAndFlags, object endSegment, int endIndexAndFlags)
         {
             // Used by SliceImpl to create new ReadOnlySequence
-            Debug.Assert(startSegment != null);
-            Debug.Assert(endSegment != null);
+            
+            // startSegment and endSegment can be null for default ReadOnlySequence
+            Debug.Assert(startSegment != null && endSegment != null || startSegment == null && endSegment == null && startIndexAndFlags == 0 && endIndexAndFlags ==  0);
```

Comment:
Is it valid for one object (say startSegment) to be null while the other is not (endSegment)?

nit: Also, can you use brackets around the condition to make it clear in what order we are doing the checks?

```C#
Debug.Assert( (startSegment != null && endSegment != null) || 
    (startSegment == null && endSegment == null && startIndexAndFlags == 0 && endIndexAndFlags ==  0) );
```

Reply:
> Is it valid for one object (say startSegment) to be null while the other is not (endSegment)?

Not it is not. startSegment and endSegment can be null only for empty sequence so startIndexAndFlags and endIndexAndFlags should be equals to 0 also.

Reply:
So, we will never have a case where startSegment is null, but endSegment is not? Either both are null (default sequence) or both are non-null?